In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [124]:
train_emnist = pd.read_csv("extended_emnist/emnist-byclass-train.csv", header=None)
test_emnist = pd.read_csv("extended_emnist/emnist-byclass-test.csv", header=None)
emnist_tr = pd.read_csv("thedataset/version2.csv", header=None)

In [125]:
emnist_tr.head()

0    1    2    3    4    5    6    7    8    9    ...  775  776  777  778  \
0   27  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0  0.0  0.0   
1   27  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0  0.0  0.0   
2   27  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0  0.0  0.0   
3   27  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0  0.0  0.0   
4   27  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  0.0  0.0  0.0  0.0   

   779  780  781  782  783  784  
0  0.0  0.0  0.0  0.0  0.0  0.0  
1  0.0  0.0  0.0  0.0  0.0  0.0  
2  0.0  0.0  0.0  0.0  0.0  0.0  
3  0.0  0.0  0.0  0.0  0.0  0.0  
4  0.0  0.0  0.0  0.0  0.0  0.0  

[5 rows x 785 columns]

In [119]:
train_emnist.head()

0    1    2    3    4    5    6    7    8    9    ...  775  776  777  778  \
0   35    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
1   36    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
2    6    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
3    3    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
4   22    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   

   779  780  781  782  783  784  
0    0    0    0    0    0    0  
1    0    0    0    0    0    0  
2    0    0    0    0    0    0  
3    0    0    0    0    0    0  
4    0    0    0    0    0    0  

[5 rows x 785 columns]

In [120]:
test_emnist.head()

0    1    2    3    4    5    6    7    8    9    ...  775  776  777  778  \
0   18    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
1   36    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
2    0    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
3    3    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   
4   33    0    0    0    0    0    0    0    0    0  ...    0    0    0    0   

   779  780  781  782  783  784  
0    0    0    0    0    0    0  
1    0    0    0    0    0    0  
2    0    0    0    0    0    0  
3    0    0    0    0    0    0  
4    0    0    0    0    0    0  

[5 rows x 785 columns]

In [121]:
len(train_emnist.columns), len(train_emnist), train_emnist.shape

(785, 697932, (697932, 785))

> This simply indicates we have 784 values which one of them is the class value. We also have '697931' of them. Which is the same as saying the size of our dataset.

In [122]:
len(test_emnist.columns), len(test_emnist), test_emnist.shape

(785, 116323, (116323, 785))

In [126]:
len(emnist_tr.columns), len(emnist_tr), emnist_tr.shape

(785, 20000, (20000, 785))

In [127]:
classes_emnist = train_emnist[0].unique()
classes_emnist.sort()

In [128]:
train_emnist, test_emnist = train_emnist.loc[train_emnist[0] >= 10 ], test_emnist.loc[test_emnist[0] >= 10 ]
print("Drop rows based on column value:\n", train_emnist)

Drop rows based on column value:
         0    1    2    3    4    5    6    7    8    9    ...  775  776  777  \
0        35    0    0    0    0    0    0    0    0    0  ...    0    0    0   
1        36    0    0    0    0    0    0    0    0    0  ...    0    0    0   
4        22    0    0    0    0    0    0    0    0    0  ...    0    0    0   
5        38    0    0    0    0    0    0    0    0    0  ...    0    0    0   
8        47    0    0    0    0    0    0    0    0    0  ...    0    0    0   
...     ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   
697926   23    0    0    0    0    0    0    0    0    0  ...    0    0    0   
697927   40    0    0    0    0    0    0    0    0    0  ...    0    0    0   
697928   47    0    0    0    0    0    0    0    0    0  ...    0    0    0   
697930   11    0    0    0    0    0    0    0    0    0  ...    0    0    0   
697931   22    0    0    0    0    0    0    0    0    0  ...    0    0    0   

     

In [129]:
classes_emnist = train_emnist[0].unique()
classes_emnist.sort()

In [130]:
classes_emnist

array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
       27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43,
       44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60,
       61], dtype=int64)

> We dropped the first 10 classes due to them being digits. I already utilized a digit based ML model.

In [158]:
mapping_emnist_tr = {27: 52, 28: 53, 67: 54, 29: 55, 68: 56, 30: 57, 31: 58, 70: 59, 32: 60, 71: 61}
emnist_tr[0] = emnist_tr[0].map(mapping_emnist_tr)
classes_emnist_tr = emnist_tr[0].unique()
classes_emnist_tr.sort()
classes_emnist_tr

array([52, 53, 54, 55, 56, 57, 58, 59, 60, 61], dtype=int64)

In [159]:
emnist_tr_train = emnist_tr.sample(frac = 0.8, random_state = 200)
emnist_tr_test = emnist_tr.drop(emnist_tr_train.index)
emnist_tr_test = emnist_tr_test.sample(frac=1, random_state=200).reset_index(drop=True)

In [160]:
len(emnist_tr_train), len(emnist_tr_test)

(16000, 4000)

In [161]:
train_emnist_last = pd.concat([train_emnist, emnist_tr_train])
test_emnist_last = pd.concat([test_emnist, emnist_tr_test])

In [162]:
len(train_emnist_last), len(test_emnist_last)

(368897, 62405)

In [163]:
classes_emnist_last = train_emnist_last[0].unique()
classes_emnist_last.sort()
classes_emnist_last

array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
       27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43,
       44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60,
       61], dtype=int64)